conans/client/settings_preprocessor.py

```diff
@@ -15,6 +15,14 @@ def _check_cppstd(settings):
     cppstd = settings.get_safe("cppstd")
     compiler_cppstd = settings.get_safe("compiler.cppstd")
 
+    if compiler == "msvc":
+        # This is hardcoded at the moment, because the minimum defined version is 19.0, with c++14
+        # TODO: When older versions of msvc are defined in settings, cppstd can be None
```

Comment:
```suggestion
        # TODO: Decide if we want to force the user to provide the 'cppstd' value always
```

Reply:
I have better removed the None value in settings leaving: ``cppstd: [14, 17, 20]``. Not defining it will raise an error. Current design was especulative, assuming older VS versions. For the current versions ``cppstd: [14, 17, 20]`` is the correct definition, lets wait to handed this in the settings_preprocessor until we really have the problem because defining older versions.

In that case we might want to add something like ``cppstd: [undefined, 14, 17, 20]`` instead of ``None``, in this way, it is still explicit and doesn't allow not defining it. The internal helpers like ``cppstd_flag`` should account for this, but this can wait.